http://sourceware.org/ml/gdb-cvs/2012-08/msg00085.html

### src/gdb/ChangeLog	2012/08/10 05:03:07	1.14574
### src/gdb/ChangeLog	2012/08/10 18:55:16	1.14575
## -1,3 +1,9 @@
+2012-08-10  Siddhesh Poyarekar  <siddhesh@redhat.com>
+
+	* python/py-type.c (convert_field): Use gdb_py_long_from_longest
+	for TYPE_FIELD_BITPOS.
+	(typy_get_sizeof): Likewise for TYPE_LENGTH.
+
 2012-08-10  Mike Frysinger  <vapier@gentoo.org>
 
 	PR cli/10436:
--- src/gdb/python/py-type.c	2012/05/18 21:02:52	1.39
+++ src/gdb/python/py-type.c	2012/08/10 18:55:18	1.40
@@ -176,7 +176,7 @@
 	}
       else
 	{
-	  arg = PyLong_FromLong (TYPE_FIELD_BITPOS (type, field));
+	  arg = gdb_py_long_from_longest (TYPE_FIELD_BITPOS (type, field));
 	  attrstring = "bitpos";
 	}
 
@@ -683,7 +683,7 @@
     }
   /* Ignore exceptions.  */
 
-  return PyLong_FromLong (TYPE_LENGTH (type));
+  return gdb_py_long_from_longest (TYPE_LENGTH (type));
 }
 
 static struct type *
